pandas/core/indexing.py

```diff
@@ -1746,7 +1746,12 @@ def _setitem_with_indexer_frame_value(self, indexer, value: "DataFrame"):
 
         unique_cols = value.columns.is_unique
 
-        if not unique_cols and value.columns.equals(self.obj.columns):
+        if name == "iloc":
+            for index, loc in zip(range(len(ilocs)), ilocs):
```

Comment:
can you call `index` something like `i` to prevent confuson with Index objects

actually use enumerate(ilocs) while we're at it

Reply:
Thx, my construct was pretty bad